### PR TITLE
Update webpack

### PR DESCRIPTION
**Pull request by ehassed3**

### CODE REVIEW

File: src/ui.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ async function startUi() {
   const store = createUiStore();
 
   store.dispatch(setLangs(LANGS));
+  console.log();
```
Comment (faergeek) [inline src/ui.tsx:41]: Should be removed I guess)

File: src/assets/amountInput.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import * as React from 'react';
 import { useIMask } from 'react-imask';
 import { BalanceAssets } from 'ui/reducers/updateState';
 import { useAppSelector } from 'ui/store';
-import * as styles from './amountInput.module.css';
+import styles from './amountInput.module.css';
```
Comment (faergeek) [inline src/assets/amountInput.tsx:7]: May be enable `namedExport` for css-loader instead?

File: package.json [hunk 3]
```diff
@@ -1,11 +1,25 @@
 {
   "private": true,
   "version": "2.9.0",
-  "engines": {
-    "node": "^10.24.1"
-  },
+  "browserslist": [
+    "maintained node versions",
```
Comment (faergeek) [inline package.json:5]: We don't want to compile for node at all

File: package.json [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+    "supports es6-module and > 0.5%",
+    "not dead",
+    "not IE 11",
```
Comment (faergeek) [inline package.json:8]: I think everything below "not IE11" is unnecessary. May be just ignore safari explicitly. You can see which browser versions are targeted by adding `"debug": true` to `babel-preset-env` options.

File: webpack.common.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
+const path = require('path');
```
Comment (faergeek) [inline webpack.common.js:1]: I don't think it's a good idea to duplicate most parts of configs. They differ not so much and it makes it hard to understand what's going on. Plus there's just to much duplication between them now. I would put all config in just one file and tweak based on `mode` passed. @smelnikov what do you think?

File: src/lib/createStreamSink.js [hunk 6]
```diff
@@ -25,10 +25,10 @@ function promiseToCallback(promise) {
   return function (cb) {
     promise.then(
       function (data) {
-        setImmediate(cb, null, data);
+        setTimeout(cb, 0, null, data);
```
Comment (faergeek) [inline src/lib/createStreamSink.js:28]: `setTimeout` is not the same as `setImmedate`. Actually I think none of them are needed here anyway. `Promise.then(...)` already creates a microtask. The same in `dnode-util`

File: package.json [hunk 7]
```diff
@@ -99,37 +115,40 @@
...
+    "mini-css-extract-plugin": "^2.6.1",
+    "mini-svg-data-uri": "^1.4.4",
     "mocha": "^9.2.2",
     "prettier": "^2.4.1",
     "selenium-webdriver": "^4.1.2",
-    "source-map-loader": "^1.0.0",
-    "style-loader": "^0.22.1",
-    "stylelint": "^13.13.1",
+    "source-map-loader": "^4.0.0",
+    "ssestream": "^1.1.0",
```
Comment (faergeek) [inline package.json:139]: Was it meant only for implementing HMR?

**Review by faergeek — CHANGES_REQUESTED** [review]

File: src/assets/amountInput.tsx (hunk 2 above)
Comment (ehassed3) [reply src/assets/amountInput.tsx:7]: I don't think this is a good idea. It is better for styles to have one name.

Comment (faergeek) [reply src/assets/amountInput.tsx:7]: It's still one name, it will just export the same thing as before, so these changes would be unnecessary

Comment (ehassed3) [reply src/assets/amountInput.tsx:7]: But it's already fixed (:

Comment (faergeek) [reply src/assets/amountInput.tsx:7]: Okay, let's leave it changed then)

File: src/lib/dnode-util.js [hunk 8]
```diff
@@ -62,7 +62,7 @@ export function promiseToCb(fn, context) {
       callback = noop;
     }
     fn.apply(context, args)
-      .then(result => setImmediate(callback, null, result))
-      .catch(error => setImmediate(callback, error));
+      .then(result => setTimeout(callback, 0, null, result))
```
Comment (faergeek) [inline src/lib/dnode-util.js:65]: `setTimeout` is not needed here as well

**Review by faergeek — APPROVED** [review]

File: src/lib/createStreamSink.js [hunk 9]
```diff
@@ -13,23 +13,13 @@ class AsyncWritableStream extends WritableStream {
...
-    throw new TypeError('Expected a promise');
-  }
-
-  return function (cb) {
-    promise.then(
-      function (data) {
-        setImmediate(cb, null, data);
+    this._asyncWriteFn(chunk, encoding).then(
+      () => {
+        callback();
```
Comment (smelnikov) [inline src/lib/createStreamSink.js:18]: Looks like you missed `null` and `data` parameters for `callback` function

**Review by smelnikov — APPROVED** [review]

File: src/global.d.ts [hunk 10]
```diff
@@ -3,6 +3,8 @@ declare const __SENTRY_DSN__: string | undefined;
 declare const __SENTRY_ENVIRONMENT__: string | undefined;
 declare const __SENTRY_RELEASE__: string | undefined;
 
+declare module '*.module.css';
```
Comment (faergeek) [inline src/global.d.ts:6]: Why is this needed?
